src/main.test.ts

```diff
@@ -211,30 +215,85 @@ function testActionOutput(actionJSPath: string, deployPath: string) {
   // If the new GITHUB_OUTPUT variable and file doesn't exist, then the core library
   // continues to fallback to this stdout approach. Since it is easy to test stdout we just
   // make sure this variable isn't set even when this test is running in GitHub actions
-  process.env.GITHUB_OUTPUT = "";
+  const env = {
+    ...environment,
+    GITHUB_OUTPUT: ""
+  };
 
-  const options: cp.ExecFileSyncOptions = {
-    env: process.env,
+  const options: childProcess.ExecFileSyncOptions = {
+    env,
   };
 
-  const result = cp.execFileSync(np, [actionJSPath], options).toString();
+  try {
+    const result = childProcess.execFileSync(np, [actionJSPath], options).toString();
 
-  const githubCommands = result.split("\n").filter((line) => line.startsWith("::"));
-  expect(githubCommands).toMatchObject(
-    [ `::set-output name=deployPath::${deployPath}` ]
-  );
+    const githubCommands = result.split("\n").filter((line) => line.startsWith("::"));
+    expect(githubCommands).toMatchObject(
+      [ `::set-output name=deployPath::${deployPath}` ]
+    );
+  } catch (error) {
+    if ((error as any).stdout) {
+      console.log("Error running command\n", (error as any).stdout.toString());
+    } else {
+      console.log("Error running command", error);
+    }
+    fail("Error running command");
+  }
 }
 
-// Test the main action using the env / stdout protocol
-test("main action runs", () => {
-  process.env.GITHUB_REF = "refs/heads/test-branch";
-  const ip = path.join(__dirname, "..", "dist", "index.js");
-  testActionOutput(ip, "branch/test-branch");
-});
+// This test runs against the built version of the code so you need to run
+// `npm run build` and `npm run package` first 
+describe("built actions run using env / stdout protocol", () => {
+  describe("main action", () => {
+    const indexPath = path.join(__dirname, "..", "dist", "index.js");
+    test("running in a push event on a branch", () => {
+      testActionOutput(
+        indexPath, 
+        {
+          // The path is needed to so the build command below will run
+          PATH: process.env.PATH!,
+          // Pass the build input is parameter
+          INPUT_BUILD: "echo no build",
+          GITHUB_REF: "refs/heads/test-branch",
+        },
+        "branch/test-branch");
+    });
 
-// Test the deploy-path action using the env / stdout protocol
-test("deploy-path action runs", () => {
-  process.env.GITHUB_REF = "refs/heads/test-branch2";
-  const ip = path.join(__dirname, "..", "deploy-path", "dist", "index.js");
-  testActionOutput(ip, "branch/test-branch2");
+    test("running in a pull_request event on a branch", () => {
+      testActionOutput(
+        indexPath, 
+        {
+          // The path is needed to so the build command below will run
+          PATH: process.env.PATH!,
+          // Pass the build input is parameter
```

Comment:
Again, sorry to be so picky, but I want to be sure I understand correctly. Should this be "...build input as parameter"?